dev/com.ibm.ws.repository.resolver/src/com/ibm/ws/repository/resolver/RepositoryResolver.java

```diff
@@ -906,7 +907,21 @@ private void reportErrors() throws RepositoryResolutionException {
         Set<ProductRequirementInformation> missingProductInformation = new HashSet<>();
 
         for (ApplicableToProduct esa : resourcesWrongProduct) {
-            missingRequirements.add(new MissingRequirement(esa.getAppliesTo(), (RepositoryResource) esa));
+            String appliesTo = esa.getAppliesTo();
+            if (appliesTo.contains("productEdition=Open")) { //Check if there are WebSphere editions and add
+                if (esa instanceof EsaResource && ((EsaResource) esa).getAppliesToFilterInfo() != null) {
+                    for (AppliesToFilterInfo atfi : ((EsaResource) esa).getAppliesToFilterInfo()) {
+                        if (!atfi.getEditions().toString().isEmpty()) {
+                            StringBuffer sb = new StringBuffer("; editions=\"");
+                            sb.append(atfi.getEditions().toString().replace("[", "").replace("]", "").replace(" ", ""));
```

Comment:
```suggestion
                            sb.append(atfi.getEditions().toString().replace("[", "").replace("]", "").replace(" ", ""));
```
sb.append(atfi.getEditions().stream().map(String::strip).collect(Collectors.joining(","));